client/src/main/java/io/pravega/client/stream/impl/SegmentSelector.java

```diff
@@ -176,8 +179,11 @@ void removeSegmentWriter(Segment segment) {
     private void createMissingWriters(Consumer<Segment> segmentSealedCallBack, String delegationToken) {
         for (Segment segment : currentSegments.getSegments()) {
             if (!writers.containsKey(segment)) {
+                DelegationTokenProvider tokenProvider = DelegationTokenProviderFactory.create(delegationToken,
```

Comment:
This should be passed in rather than constructed here.

Reply:
Sorry, I didn't understand how constructing the `DelegationTokenProvider` object in the calling method help here. Could you please explain your comment?

Reply:
I've modified this class to have a `DelegationTokenProvider` object as instance variable, which is in turn populated with the latest delegation token that is received. So, this issue is now resolved.